var/spack/repos/builtin/packages/emacs/package.py

```diff
@@ -36,8 +36,12 @@ class Emacs(AutotoolsPackage):
     version('24.5', 'd74b597503a68105e61b5b9f6d065b44')
 
     variant('X', default=False, description="Enable an X toolkit")
-    variant('toolkit', default='gtk',
-            description="Select an X toolkit (gtk, athena)")
+    variant(
+        'toolkit',
+        default='gtk',
+        values=('gtk', 'athena'),
+        description="Select an X toolkit (gtk, athena)"
+    )
```

Comment:
I can fix some of these.  Err, ok, charm and texlive are easy.

One of `matlab`'s variants is "key", which has an empty default value and is presumably the license key.  What would the `values()` for that be?

`npb` has several variants for which legal values are comma separated strings composed from subsets of lists of 7-9 things

`plumed` has `optional_modules`, which is a "String that is used to build optional modules", with a default value of "all".  I'm not sure what the legal options are.

It almost feels like there needs to be a `values_regexp(...)` or something to valid a variant w/out requiring that all possibilities be listed.  Or better yet (is this easy in python?), a function (reference to a function, ...) to use to validate a variant value.